action.js

```diff
@@ -1,10 +1,96 @@
-chrome.browserAction.onClicked.addListener(() => {
-    chrome.tabs.query({
-        active: true,
-        currentWindow: true
-    }, function (tabs) {
-        chrome.tabs.sendMessage(tabs[0].id, {
-            message: "host"
-        });
+// We monitor all `<iframe` and top-frame content for a `<video` tag, and store
+// the frame info to be used when our chrome.browserAction is clicked.
+let videoFrame;
```

Comment:
I have a feeling that `videoFrame` is going to retain the old frame even after the user closes the tab/navigates away. That probably causes the extension to crash if you click the Extension button on a page with no video.

Maybe should address that (?)

Reply:
This was true. If your tab ever shows a video and then stops, it would host a session and get into a weird state.

It's pretty hard to know when to clear this value, so I'm just gonna throw a vague error if you press the button